### PR TITLE
Bug 1878868 - Create TestSetup helper: Test classes A-C

### PR DESCRIPTION
**Pull request by sv-ohorvath**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1878868

### CODE REVIEW

**Review by AndiAJ — APPROVED** [review]
👍🏽

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestSetup.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            // Reset locale to EN-US if needed.
+            AppAndSystemHelper.resetSystemLocaleToEnUS()
+            // Check and clear the downloads folder
+            AppAndSystemHelper.clearDownloadsFolder()
+            // Make sure the Wifi and Mobile Data connections are on
+            AppAndSystemHelper.setNetworkEnabled(true)
+            // Clear bookmarks left after a failed test
+            val bookmarks = bookmarksStorage.getTree(BookmarkRoot.Mobile.id)?.children
+            Log.i(TAG, "Before cleanup: Bookmarks storage contains: $bookmarks")
+            bookmarks?.forEach { bookmarksStorage.deleteNode(it.guid) }
```
Comment (jjSDET) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestSetup.kt:34]: try using a `suspend coroutine` here, and let's see if this fixes it. it _should_ only continue after we get confirmation the `delete` instruction as been commited and applied to the data store. 🤞 

```
import kotlin.coroutines.resume
import kotlin.coroutines.resumeWithException
import kotlin.coroutines.suspendCoroutine

private suspend fun deleteBookmark(guid: String) {
        suspendCoroutine<Unit> { continuation ->
            bookmarksStorage.deleteNode(guid) { success ->
                if (success) {
                    continuation.resume(Unit)
                } else {
                    continuation.resumeWithException(Exception("Failed to delete bookmark with GUID $guid"))
                }
            }
        }
    }
```

**Review by jjSDET — CHANGES_REQUESTED** [review]

**Review by jjSDET — APPROVED** [review]
We'll keep looking into finding some good ways to verify the db status, but for now, this looks good. 

We can hack more on this later this week and get something in place early next week.

**Comment by AaronMT** [comment]
Reverting as this breaks `ui-test-apk-legacy` tests on 26-29 due to missing methods in some of the Android calls. If you'd like to test that Task on a PR, [modify this line](https://github.com/mozilla-mobile/firefox-android/blob/b6efccdd0df0d50c391b2b5671f776665c24e8e0/taskcluster/ci/ui-test-apk/kind.yml#L410) to `github-pull-request`

**Pull request merged**